Applications/Spire/Include/Spire/OrderImbalanceIndicator/LocalOrderImbalanceIndicatorModel.hpp

```diff
@@ -0,0 +1,37 @@
+#ifndef SPIRE_LOCAL_ORDER_IMBALANCE_INDICATOR_MODEL_HPP
+#define SPIRE_LOCAL_ORDER_IMBALANCE_INDICATOR_MODEL_HPP
+#include "Spire/OrderImbalanceIndicator/OrderImbalanceIndicatorModel.hpp"
+
+namespace Spire {
+
+  //! Represents an OrderImbalanceIndicatorModel that accepts insertion of
+  //! new order imbalances.
+  class LocalOrderImbalanceIndicatorModel :
+      public OrderImbalanceIndicatorModel {
+    public:
+
+      //! Inserts an order imbalance into the model.
+      /*
+        \param imbalance The order imbalance to insert into the model.
+      */
+      void insert(const Nexus::OrderImbalance& imbalance);
+
+      std::tuple<boost::signals2::connection,
```

Comment:
Don't use a tuple, in ```OrderImbalanceIndicatorModel``` define a ```struct``` with fields to represent the connection and the snapshot.